tests/integ_tests/main.rs

```diff
@@ -607,6 +611,47 @@ async fn test_http_content_encoding_suffix() {
     assert_eq!(json_data.to_owned(), body_to_string(response).await);
 }
 
+#[tokio::test]
+async fn test_http_authorization_source() {
+    // Start app server
+    let app_server = MockServer::start();
+    let hello = app_server.mock(|when, then| {
+        when.method(GET).path("/hello").header_exists("Authorization");
```

Comment:
`header_exists` makes requests fail if it does not contain Authorization header. https://docs.rs/httpmock/latest/httpmock/struct.When.html#method.header_exists

This effectively validates the authorization_source behavior.